cache/internal/profile/cache.go

```diff
@@ -0,0 +1,110 @@
+package profile
+
+import (
+	"cache/internal/order"
+	"fmt"
+	"sync"
+	"time"
+)
+
+type Cache struct {
+	mu   sync.RWMutex
+	data map[string]*CacheItem
+	ttl  time.Duration
+}
+
+type CacheItem struct {
+	profile  *Profile
+	expireAt time.Time
+}
+
+func NewCache(ttl time.Duration) *Cache {
+	cache := &Cache{
+		data: make(map[string]*CacheItem),
+		ttl:  ttl,
+		mu:   sync.RWMutex{},
+	}
+	go cache.cleanup()
+	return cache
+}
+
+func (c *Cache) Get(uuid string) *Profile {
+	c.mu.RLock()
+	defer c.mu.RUnlock()
+
+	p, ok := c.data[uuid]
+
+	if !ok {
+		return nil
+	}
+
+	if time.Now().After(p.expireAt) {
+		return nil
+	}
+
+	return p.profile
```

Comment:
you seem to return a reference to profile which if changed - will change in the cache aswell

Reply:
Thanks, fixed